Index: APM_BinComm/APM_BinComm.cpp
===================================================================
--- APM_BinComm/APM_BinComm.cpp	(revision 741)
+++ APM_BinComm/APM_BinComm.cpp	(working copy)
@@ -47,7 +47,9 @@
 BinComm::BinComm(const BinComm::MessageHandler *handlerTable,
                  Stream *interface) :
         _handlerTable(handlerTable),
-        _interface(interface)
+        _interface(interface),
+		_lastReceived(millis()),
+		_decodePhase(DEC_WAIT_P1)
 {
 };
 
@@ -101,7 +103,10 @@
         // handle inter-byte timeouts (resync after link loss, etc.)
         //
         if ((millis() - _lastReceived) > DEC_MESSAGE_TIMEOUT)
-                _decodePhase = DEC_WAIT_P1;
+		{
+				_lastReceived = millis();
+        		_decodePhase = DEC_WAIT_P1;
+		}
 
         // run the decode state machine
         //
@@ -191,6 +196,7 @@
                 if (inByte == _sumB) {
                         // if we got this far, we have a message
                         messagesReceived++;
+						_lastReceived = millis();
 
                         // call any handler interested in this message
                         for (tableIndex = 0; MSG_NULL != _handlerTable[tableIndex].messageID; tableIndex++)
@@ -202,5 +208,7 @@
                 }
                 _decodePhase = DEC_WAIT_P1;
                 break;
+		default:
+				_decodePhase = DEC_WAIT_P1;		
         }
 }
